src/Framework/Hooks/CommonConditions.cs

```diff
@@ -123,67 +126,145 @@ private static QuestStatSummary GetQuestStats(CustomQuest managedQuest)
+                int expectedFriendshipLevel = Convert.ToInt32(flevel[i + 1]);
+
+                flag &= currentFriendshipLevel >= expectedFriendshipLevel;
+
+                if (Monitor.IsVerbose)
+                    Monitor.Log(
+                        $"Checked friendship level for `{whoLevel}`, " +
+                        $"current level: {currentFriendshipLevel}, " +
+                        $"expected: {expectedFriendshipLevel}, " +
+                        $"current flag: {flag}");
+            }
+
+            return flag;
+        }
+
+        public static bool CheckFriendshipStatus(string friendshipStatus) 
+        {
+            string[] fstatus = friendshipStatus.Split(' ');
+            bool flag = true;
+
+            if (fstatus.Length < 2)
+                return false;
+
+            for (int i = 0; i < fstatus.Length; i += 2)
+            {
+                string whoStatus = fstatus[i];
+                string currentStatus = Game1.player.friendshipData[whoStatus].Status.ToString();
+                string expectedStatus = fstatus[i + 1];
+
+                flag &= currentStatus == expectedStatus;
+
+                if (Monitor.IsVerbose)
+                    Monitor.Log(
+                        $"Checked friendship status for `{whoStatus}`, " +
+                        $"current status: {currentStatus}, " +
+                        $"expected status: {expectedStatus}, " +
+                        $"current flag: {flag}");
+            }
+            
+            return flag;
+        }
+
+        public static bool CheckSkillLevel(string skillLevel) 
+        {
+            string[] slevel = skillLevel.Split(' ');
+            bool flag = true;
+
+            if (slevel.Length < 2)
+                return false;
+
+            for (int i = 0; i < slevel.Length; i += 2)
+            {
+                string[] skillList = { "Farming", "Fishing", "Foraging", "Mining", "Combat", "Luck" };
+                string skillName = slevel[i];
+                int skillId = Array.IndexOf(skillList, slevel[i]);
+                int currentSkillLevel = Game1.player.getEffectiveSkillLevel(skillId);
+                int expectedSkillLevel = Convert.ToInt32(slevel[i + 1]);
+
+                flag &= currentSkillLevel >= expectedSkillLevel;
+
+                if (Monitor.IsVerbose)
+                    Monitor.Log(
+                        $"Checked skill level for `{skillName}`, " +
+                        $"with skill id: {skillId}, " +
+                        $"current level: {currentSkillLevel}, " +
+                        $"expected level: {expectedSkillLevel}, " +
+                        $"current flag: {flag}");
+            }
+
+            return flag;
+        }
+
+        public static bool CheckBuilding(string checkedBuilding)
+        {
+            checkedBuilding = checkedBuilding.Replace(", ", "|");
+            string[] building = checkedBuilding.Split('|');

```

Comment:
Keep space here in split, see next comment's suggestion.

```suggestion
            string[] building = checkedBuilding.Split(' ');
```